great_expectations/checkpoint/checkpoint.py

```diff
@@ -407,8 +408,41 @@ def _run_validation(
                 f"Exception occurred while running validation[{idx}] of Checkpoint '{self.name}': {e.message}."
             ) from e
 
-    def self_check(self, pretty_print=True) -> dict:
-        # Provide visibility into parameters that Checkpoint was instantiated with.
+    @public_api
+    def self_check(self, pretty_print: bool = True) -> dict:
+        """
+        Method that is intended to provide visibility into parameters that Checkpoint was instantiated with.
+
+        If used as part of the test_yaml_config() workflow, the user will be able to know if the Checkpoint is
+        configured with all necessary parameters.
+
+        When run with self_check()::
+
+            yaml_config: str = # [a checkpoint yaml configuration]
+            config: CommentedMap = yaml.load(yaml_config)
+            checkpoint_config: CheckpointConfig = CheckpointConfig(**config)
+            checkpoint: Checkpoint = Checkpoint(
+                data_context=context,
+                checkpoint_config.to_json_dict()
+            )
+            checkpoint.self_check()
+
+
+        When run with test_yaml_config::
+
+            checkpoint: Checkpoint = context.test_yaml_config(
+                yaml_config=yaml_config,
+                name="my_checkpoint"
+                )
+
+        Args:
+            pretty_print (bool): shows how the print additional messages if Checkpoint configuration is missing
+            a "validations" or "action_list" attribute.
```

Comment:
The arg description looks a little off here

Reply:
sorry about this one. Editing didn't get pushed